tests/garage/torch/algos/test_vpg.py

```diff
@@ -0,0 +1,103 @@
+from tests.fixtures import snapshot_config
+
+
+class TestVPG:
+
+    @classmethod
+    def setup_class(cls):
+        deterministic.set_seed(0)
+
+    def setup_method(self):
+        self._env = GarageEnv(gym.make('InvertedDoublePendulum-v2'))
+        self._runner = LocalRunner(snapshot_config)
+
+        policy = GaussianMLPPolicy(env_spec=self._env.spec,
+                                   hidden_sizes=[64, 64],
+                                   hidden_nonlinearity=torch.tanh,
+                                   output_nonlinearity=None)
+        self._params = {
+            'env_spec': self._env.spec,
+            'policy': policy,
+            'optimizer': torch.optim.Adam,
+            'baseline': LinearFeatureBaseline(env_spec=self._env.spec),
+            'max_path_length': 100,
+            'discount': 0.99,
+            'policy_lr': 1e-2
+        }
+
+    def teardown_method(self):
+        self._env.close()
+
+    def test_vpg_no_entropy(self):
+        """Test VPG with no_entropy."""
+        self._params['positive_adv'] = True
+        self._params['use_softplus_entropy'] = True
+
+        algo = VPG(**self._params)
+        self._runner.setup(algo, self._env)
+        last_avg_ret = self._runner.train(n_epochs=10, batch_size=100)
+        assert last_avg_ret > 30
+
+    def test_vpg_max(self):
+        """Test VPG with maximum entropy."""
+        self._params['center_adv'] = False
+        self._params['stop_entropy_gradient'] = True
+        self._params['entropy_method'] = 'max'
+
+        algo = VPG(**self._params)
+        self._runner.setup(algo, self._env)
+        last_avg_ret = self._runner.train(n_epochs=10, batch_size=100)
+        assert last_avg_ret > 30
+
+    def test_vpg_regularized(self):
+        """Test VPG with entropy_regularized."""
+        self._params['entropy_method'] = 'regularized'
+
+        algo = VPG(**self._params)
+        self._runner.setup(algo, self._env)
+        last_avg_ret = self._runner.train(n_epochs=10, batch_size=100)
+        assert last_avg_ret > 30
+
+    def test_invalid_entropy_method(self):
+        self._params['entropy_method'] = 'INVALID_ENTROPY_METHOD'
+
+        expected_msg = 'Invalid entropy_method'
+        with pytest.raises(ValueError, match=expected_msg):
+            VPG(**self._params)
+
+    def test_invalid_max_entropy_config_with_center_adv(self):
+        self._params['entropy_method'] = 'max'
+        self._params['center_adv'] = True
+
+        expected_msg = 'center_adv should be False when entropy_method is max'
+        with pytest.raises(ValueError, match=expected_msg):
+            VPG(**self._params)
+
+    # yapf: disable
+    def test_invalid_max_entropy_config_with_stop_entropy_gradient(self):
+        self._params['entropy_method'] = 'max'
+        self._params['center_adv'] = False
+        self._params['stop_entropy_gradient'] = False
+
+        expected_msg = 'stop_gradient should be True when entropy_method is max'  # noqa: E501
```

Comment:
you can match only a portion of the message to make sure it's relevant (e.g. `'stop_gradient'`), rather than the entire message. this makes less work for someone who wants to change the message a little.

Reply:
fixed!

Reply:
i don't see the changes. did you forget to push?